frontend/src/lib/modals/neurons/LosingRewardNeuronsModal.svelte

```diff
@@ -8,15 +8,30 @@
   import { soonLosingRewardNeuronsStore } from "$lib/derived/neurons.derived";
   import NnsLosingRewardsNeuronCard from "$lib/components/neurons/NnsLosingRewardsNeuronCard.svelte";
   import { listKnownNeurons } from "$lib/services/known-neurons.services";
+  import { goto } from "$app/navigation";
+  import { buildNeuronUrl } from "$lib/utils/navigation.utils";
+  import { OWN_CANISTER_ID_TEXT } from "$lib/constants/canister-ids.constants";
+  import type { NeuronInfo } from "@dfinity/nns";
 
-  const dispatcher = createEventDispatcher();
+  const dispatcher = createEventDispatcher<{ nnsClose: void }>();
 
   // Load KnownNeurons which are used in the FollowNnsTopicSections
   onMount(() => listKnownNeurons());
 
   const confirm = () => {
     // TBD
   };
+  const close = () => dispatcher("nnsClose");
+
+  const navigateToNeuronDetail = async (neuron: NeuronInfo) => {
+    close();
+    goto(
```

Comment:
you could introduce  a tick of svelte. it would allow it to catch up with stuff to do and the do the navigation. 

```suggestion
  await tick();
  goto(buildNeuronUrl({
     universe: OWN_CANISTER_ID_TEXT,
      neuronId: neuron.neuronId,
  })
```

Reply:
Thank you for mentioning this. But in this case, the current solution works, and I’d prefer to avoid adding code just for precaution.